File: java/bank-simulation/src/main/java/com/nursyah/bank_simulation/features/customer/CustomerController.java
package com.nursyah.bank_simulation.features.customer;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import com.nursyah.bank_simulation.features.customer.dto.RequestSendMoney;
import com.nursyah.bank_simulation.features.customer.dto.RequestWidthdrawal;
import com.nursyah.bank_simulation.shared.dto.RequestPaginated;
import com.nursyah.bank_simulation.shared.dto.ResponseDto;
import com.nursyah.bank_simulation.shared.dto.ResponsePaginated;
import com.nursyah.bank_simulation.shared.dto.ResponseTransaction;

import java.util.concurrent.CompletableFuture;

@RestController
@RequestMapping("/api/v1/customer")
public class CustomerController {
    private final ICustomerService customerService;

    @Autowired
    public CustomerController(ICustomerService customerService) {
        this.customerService = customerService;
    }

    @PostMapping("/send")
    public CompletableFuture<ResponseEntity<ResponseDto>> sendMoney(@RequestBody RequestSendMoney request) {
        return customerService.sendMoneyAsync(request).thenApply(ResponseEntity::ok);
    }

    @PostMapping("/withdrawal")
    public CompletableFuture<ResponseEntity<ResponseDto>> withdrawalMoney(@RequestBody RequestWidthdrawal request) {
        return customerService.withdrawalMoneyAsync(request).thenApply(ResponseEntity::ok);
    }

    @GetMapping("/transaction")
    public CompletableFuture<ResponseEntity<ResponsePaginated<ResponseTransaction>>> viewMyTransactions(@RequestParam RequestPaginated request) {
        return customerService.viewMyTransactionsAsync(request).thenApply(ResponseEntity::ok);
    }
}

File: dotnet/bank_simulation/Features/Customer/CustomerController.cs
using bank_simulation.Features.Customer.Dto;
using bank_simulation.Shared.Dto;
using Microsoft.AspNetCore.Mvc;

namespace bank_simulation.Features.Customer;

[ApiController]
[Route("api/v1/[controller]")]
public class CustomerController : ControllerBase
{
    private readonly ICustomerService _customerService;
    public CustomerController(ICustomerService customerService)
    {
        _customerService = customerService;
    }

    [HttpPost("send")]
    public async Task<IActionResult> SendMoney([FromBody] RequestSendMoney request)
    {
        var result = await _customerService.SendMoneyAsync(request);
        return Ok(result);
    }

    [HttpPost("withdrawal")]
    public async Task<IActionResult> WithdrawalMoney([FromBody] RequestWidthdrawal request)
    {
        var result = await _customerService.WithdrawalMoneyAsync(request);
        return Ok(result);
    }

    [HttpGet("transaction")]
    public async Task<IActionResult> ViewMyTransactions([FromQuery] RequestPaginated request)
    {
        var result = await _customerService.ViewMyTransactionsAsync(request);
        return Ok(result);
    }
}
